datafusion/functions/src/math/log.rs

```diff
@@ -283,4 +289,52 @@ mod tests {
             }
         }
     }
+    #[test]
+    // Test log() simplification errors
+    fn test_log_simplify_errors() {
+        let props = ExecutionProps::new();
+        let schema =
+            Arc::new(DFSchema::new_with_metadata(vec![], HashMap::new()).unwrap());
+        let context = SimplifyContext::new(&props).with_schema(schema);
+        // Expect 0 args to arror
+        let _ = LogFunc::new().simplify(vec![], &context).unwrap_err();
+        // Expect 3 args to error
+        let _ = LogFunc::new()
+            .simplify(vec![lit(1), lit(2), lit(3)], &context)
+            .unwrap_err();
+    }
+
+    #[test]
+    // Test that non-simplifiable log() expressions are unchanged after simplification
+    fn test_log_simplify_original() {
+        let props = ExecutionProps::new();
+        let schema =
+            Arc::new(DFSchema::new_with_metadata(vec![], HashMap::new()).unwrap());
+        let context = SimplifyContext::new(&props).with_schema(schema);
+        // One argument with no simplifications
+        let result = LogFunc::new().simplify(vec![lit(2)], &context).unwrap();
+        match result {
+            ExprSimplifyResult::Simplified(_) => {
+                panic!("Expected ExprSimplifyResult::Original")
+            }
+            ExprSimplifyResult::Original(args) => {
+                assert_eq!(args.len(), 1);
+                assert_eq!(args[0], lit(2));
+            }
+        }
```

Comment:
Another way to write this perhaps more concisely and with less indenting could be: 


```suggestion
        let ExprSimplifyResult::Original(args) = result  else {
            panic!("Expected ExprSimplifyResult::Original")
        };
        assert_eq!(args.len(), 1);
        assert_eq!(args[0], lit(2));
```

Reply:
(same for the other)

Reply:
tbh I always forget `let else` exists 😆